civihr_employee_portal/src/Mail/EmergencyContactDeletionNotificationMail.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Drupal\civihr_employee_portal\Mail;
+
+use Drupal\civihr_employee_portal\Service\ContactService;
+
+class EmergencyContactDeletionNotificationMail extends AbstractDrupalSystemMail {
+
+  /**
+   * @inheritdoc
+   */
+  public function getVariables($message) {
+    $contactId = \CRM_Core_Session::singleton()->getLoggedInContactID();
+    $contact = civicrm_api3('Contact', 'getsingle', ['id' => $contactId]);
+    $contactEmail = ContactService::getContactWorkEmail($contactId);
+    $profileLink = ContactService::getLinkToContactProfile($contactId);
+    $emergencyContact = $message['params']['emergencyContact'];
+
+    return [
+      'workEmail' => $contactEmail,
+      'profileLink' => $profileLink,
+      'displayName' => $contact['display_name'],
+      'submissionDate' => date('Y-m-d H:i'),
+      'emergencyContactName' => $emergencyContact['Name'],
+      'emergencyContactType' => $this->getEmergencyContactType($message),
+    ];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function getTemplateName() {
+    return 'emergency_contact_deletion.tpl';
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function getSubject($message) {
+    $format = 'CiviHR Self Service Data Submission - %s Deleted';
+
+    return sprintf($format, ucwords($this->getEmergencyContactType($message)));
+  }
+
+  /**
+   * Gets the emergency contact from the message and determines if it is a
+   * 'dependant' or 'emergency contact'
+   *
+   * @param array $message
+   *
+   * @return string
+   */
+  private function getEmergencyContactType($message) {
```

Comment:
Looks like you only really need the emergency contact here. How about passing only that instead of the entire message?